### PR TITLE
Feature/kak 115 change catalog structure

### PR DESCRIPTION
**Pull request by DosiaTG**
- Structure changes connected with the description in task KAK-115
- import changes 
- unit tests fix after structure changes
- documentation updates

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -144,37 +144,49 @@ npm run kakunin
...
-ln -s ../node_modules/kakunin/src/step_definitions/navigation.ts kakunin-navigation.ts 
-ln -s ../node_modules/kakunin/src/step_definitions/performance.ts kakunin-performance.ts 
+ln -s ../node_modules/kakunin/dist/web/step_definitions/elements.js kakunin-elements.js
+ln -s ../node_modules/kakunin/dist/web/step_definitions/navigation.js kakunin-navigation.js 
+ln -s ../node_modules/kakunin/dist/web/step_definitions/form.js kakunin-form.js
+ln -s ../node_modules/kakunin/dist/web/step_definitions/tabs.js kakunin-tabs.js
+
+ln -s ../node_modules/kakunin/dist/rest/step_definitions/api.js kakunin-api.js
+
+ln -s ../node_modules/kakunin/dist/common/step_definitions/debug.js kakunin-debug.js
```
Comment (tgorskievo) [inline docs/index.md]: `kakunin-debug.js` should be in web catalog as it contains only `browser.sleep()`

File: docs/index.md [hunk 2]
```diff
@@ -144,37 +144,49 @@ npm run kakunin
...
+ln -s ../node_modules/kakunin/dist/common/step_definitions/file.js kakunin-file.js
```
Comment (tgorskievo) [inline docs/index.md]: It should be in `common`

File: docs/index.md [hunk 3]
```diff
@@ -144,37 +144,49 @@ npm run kakunin
...
+ln -s ../node_modules/kakunin/dist/common/step_definitions/email.js kakunin-email.js
```
Comment (tgorskievo) [inline docs/index.md]: common

File: docs/index.md [hunk 4]
```diff
@@ -144,37 +144,49 @@ npm run kakunin
...
+ln -s ../node_modules/kakunin/dist/common/step_definitions/generators.js kakunin-generators.js
```
Comment (tgorskievo) [inline docs/index.md]: common as it only generate values and stores them to variableStore

File: docs/index.md [hunk 5]
```diff
@@ -144,37 +144,49 @@ npm run kakunin
...
+ln -s ../node_modules/kakunin/dist/common/step_definitions/performance.js kakunin-performance.js
```
Comment (tgorskievo) [inline docs/index.md]: I think, it should be web as we save the `.har` file generated in a browser

File: docs/index.md [hunk 6]
```diff
@@ -144,37 +144,49 @@ npm run kakunin
...
+ln -s ../node_modules/kakunin/dist/web/step_definitions/tabs.js kakunin-tabs.js
+
+ln -s ../node_modules/kakunin/dist/rest/step_definitions/api.js kakunin-api.js
+
+ln -s ../node_modules/kakunin/dist/common/step_definitions/debug.js kakunin-debug.js
+ln -s ../node_modules/kakunin/dist/common/step_definitions/file.js kakunin-file.js
+ln -s ../node_modules/kakunin/dist/common/step_definitions/email.js kakunin-email.js
+ln -s ../node_modules/kakunin/dist/common/step_definitions/generators.js kakunin-generators.js
+ln -s ../node_modules/kakunin/dist/common/step_definitions/performance.js kakunin-performance.js 
+ln -s ../node_modules/kakunin/dist/common/step_definitions/screenshots.js kakunin-screenshots.js
```
Comment (tgorskievo) [inline docs/index.md]: We take a screenshot of a browser window, so it should be web

File: docs/index.md [hunk 7]
```diff
@@ -144,37 +144,49 @@ npm run kakunin
 In order to make cucumber steps autosuggestion work properly in JetBrains tools, make sure your project is `ECMAScript 6` compatible and you have `cucumberjs` plugin installed.
 Due to non-resolved issue in Jetbrains editors ([see here](https://youtrack.jetbrains.com/issue/WEB-11505)) we'll have to do one more step:
  
-Go to `step_definitions` directory 
+Go to `step_definitions` directory for each folder
```
Comment (tgorskievo) [inline docs/index.md]: I think that the previous description was okay, as we create symlinks a new Kakunin project (not inside the core)

**Comment by simonsipl** [comment]
I have small doubts about this:
1. When my pr will be merged first it may fail CI on this PR, if this will be merged first it will fail my pr

File: docs/index.md [hunk 8]
```diff
@@ -144,37 +144,49 @@ npm run kakunin
...
 ```bash
-ln -s ../node_modules/kakunin/src/step_definitions/elements.ts kakunin-elements.ts
-ln -s ../node_modules/kakunin/src/step_definitions/debug.ts kakunin-debug.ts
-ln -s ../node_modules/kakunin/src/step_definitions/file.ts kakunin-file.ts
-ln -s ../node_modules/kakunin/src/step_definitions/form.ts kakunin-form.ts
-ln -s ../node_modules/kakunin/src/step_definitions/email.ts kakunin-email.ts
-ln -s ../node_modules/kakunin/src/step_definitions/generators.ts kakunin-generators.ts
-ln -s ../node_modules/kakunin/src/step_definitions/navigation.ts kakunin-navigation.ts 
-ln -s ../node_modules/kakunin/src/step_definitions/performance.ts kakunin-performance.ts 
+ln -s ../node_modules/kakunin/dist/web/step_definitions/elements.js kakunin-elements.js
```
Comment (tgorskievo) [inline docs/index.md]: You can move the steps to folders (web, rest, common) inside the initialised project

/step_definitions
   /rest/step.js
   /web/step.js
   /common/step.js

Comment (DosiaTG) [reply docs/index.md]: done

File: docs/index.md [hunk 9]
```diff
@@ -144,37 +144,49 @@ npm run kakunin
 In order to make cucumber steps autosuggestion work properly in JetBrains tools, make sure your project is `ECMAScript 6` compatible and you have `cucumberjs` plugin installed.
 Due to non-resolved issue in Jetbrains editors ([see here](https://youtrack.jetbrains.com/issue/WEB-11505)) we'll have to do one more step:
  
-Go to `step_definitions` directory 
+Go to `step_definitions` directory
 ```bash
-cd step_definitions
+cd step_definitions/web
```
Comment (sethii) [inline docs/index.md]: There is no web, res or common directory inside step_definitions (in our project). This should be a single directory without subdirectories.

File: docs/index.md [hunk 10]
```diff
@@ -144,37 +144,48 @@ npm run kakunin
 In order to make cucumber steps autosuggestion work properly in JetBrains tools, make sure your project is `ECMAScript 6` compatible and you have `cucumberjs` plugin installed.
 Due to non-resolved issue in Jetbrains editors ([see here](https://youtrack.jetbrains.com/issue/WEB-11505)) we'll have to do one more step:
  
-Go to `step_definitions` directory 
+Go to `step_definitions` directory
 ```bash
 cd step_definitions
+mkdir common && mkdir rest && mkdir web
```
Comment (tgorskievo) [inline docs/index.md]: I checked it, and Kakunin sees the steps in the `step_definitions/web/step.js` directory. Also, the autosuggestion works. @sethii 
So keeping it in separated folders might be better.

File: docs/index.md [hunk 11]
```diff
@@ -154,27 +154,37 @@ Paste this code into terminal and restart your IDE:
...
-ln -s ../node_modules/kakunin/src/step_definitions/navigation.ts kakunin-navigation.ts 
-ln -s ../node_modules/kakunin/src/step_definitions/performance.ts kakunin-performance.ts 
+ln -s ../node_modules/kakunin/dist/web/step_definitions/elements.js web-kakunin-elements.js
+ln -s ../node_modules/kakunin/dist/web/step_definitions/navigation.js web-kakunin-navigation.js 
+ln -s ../node_modules/kakunin/dist/web/step_definitions/form.js web-kakunin-form.js
+ln -s ../node_modules/kakunin/dist/web/step_definitions/tabs.js web-kakunin-tabs.js
+ln -s ../node_modules/kakunin/dist/web/step_definitions/debug.js web-kakunin-debug.js
+ln -s ../node_modules/kakunin/dist/web/step_definitions/performance.js web-kakunin-performance.js 
+ln -s ../node_modules/kakunin/dist/web/step_definitions/screenshots.js web-kakunin-screenshots.js 
+
```
Comment (joszo) [inline docs/index.md]: do we need those empty spaces?

Comment (DosiaTG) [reply docs/index.md]: I added this to separate sections which are going to different folders, but now if we won't split all of them on folders I can remove it

**Comment by tgorskievo** [comment]
Please squash those 9 commits :)

**Pull request merged**